docs/understand/programming_model.rst

```diff
@@ -7,164 +7,257 @@
+  Some C++ features have poor latency when implemented on GPU devices, therefore
+  they are forbidden in device contexts to avoid using features that unexpectedly
+  decimate the program's performance. Offload devices targeted by HIP aren't general
+  purpose devices, at least not in the sense that a CPU is. HIP focuses on data
+  parallel computations and as such caters to throughput optimized architectures,
+  such as GPUs or accelerators derived from GPU architectures.
 
-* Asynchrony is at the forefront of the HIP API. Computations launched on the device
+* Asynchronicity is at the forefront of the HIP API. Computations launched on the device
   execute asynchronously with respect to the host, and it is the user's responsibility to
   synchronize their data dispatch/fetch with computations on the device.
 
   .. note::
-    HIP does perform implicit synchronization on occasions, more advanced than other
-    APIs such as OpenCL or SYCL, in which the responsibility of synchronization mostly
-    depends on the user.
+    HIP performs implicit synchronization on occasions, unlike some
+    APIs where the responsibility for synchronization is left to the user.
+
+Host programming
+----------------
+
+In heterogeneous programming, the CPU is available for processing operations but the host application has the additional task of managing data and computation exchanges between the CPU (host) and GPU (device). The host acts as the application manager, coordinating the overall workflow and directing operations to the appropriate context, handles data preparation and data transfers, and manages GPU tasks and synchronization. Here is a typical sequence of operations:
+
+1.	Initialize the HIP runtime and select the GPU: As described in :ref:`initialization`, refers to identifying and selecting a target GPU, setting up a context to let the CPU interact with the GPU.  
+2.	Data preparation: As discussed in :ref:`memory_management`, this includes allocating the required memory on the host and device, preparing input data and transferring it from the host to the device. The data is both transferred to the device, and passed as an input parameter when launching the kernel. 
+3.	Configure and launch the kernel on the GPU: As described in :ref:`device_program`, define and load the kernel or kernels to be run, launch kernels using the triple chevron syntax or appropriate API call (for example ``hipLaunchKernelGGL``), and pass parameters as needed. On the GPU, kernels run on streams, or a queue of operations. Within the same stream, operations run in the order they were issued, but different streams are independent and can execute concurrently. In the HIP runtime, kernels run on the default stream when one is not specified, but specifying a stream for the kernel lets you increase concurrency in task scheduling and resource utilization, and launch and manage multiple kernels from the host program.
```

Comment:
"define and load the kernel or kernels to be run, launch kernels using the triple chevron syntax or appropriate API call (for example ``hipLaunchKernelGGL``), and pass parameters as needed. On the GPU, kernels run on streams, or a queue of operations."
Should be 
"defines kernel configurations and arguments, launches kernel to excute on the GPU device using the triple chevron syntax or appropriate API call (for example hipLaunchKernelGGL). On the GPU, multiple kernels can run on streams, with a queue of operations. "